runtime/vm/LayoutFFITypeHelpers.cpp

```diff
@@ -203,7 +203,13 @@ LayoutFFITypeHelpers::getStructFFITypeElements(char **layout)
 void
 LayoutFFITypeHelpers::freeStructFFIType(ffi_type *ffiType)
 {
-	if ((NULL != ffiType) && (FFI_TYPE_STRUCT == ffiType->type)) {
+	if ((NULL != ffiType)
+			&& ((FFI_TYPE_STRUCT == ffiType->type)
+#if defined(J9ZOS390)
```

Comment:
This also needs to check `defined(J9VM_ENV_DATA64)`, otherwise I think 31-bit z/OS builds will break.
If the new types should be supported in 31-bit builds, then `mz31/ffi.h`, etc. need to be updated.